dagshub/common/helpers.py

```diff
@@ -31,3 +40,99 @@ def http_request(method, url, **kwargs):
         if arg not in kwargs:
             kwargs[arg] = mixin_args[arg]
     return httpx.request(method, url, **kwargs)
+
+
+def get_project_root(root):
+    while not (root / '.git').is_dir():
+        if ismount(root):
+            raise ValueError('No git project found! (stopped at mountpoint {root}). \
+                             Please run this command in a git repository.')
+        root = root / '..'
+    return Path(root)
+
+
+def _init(repo_name=None, repo_owner=None, url=None, root=None,
+          host=config.DEFAULT_HOST, mlflow=True, dvc=False):
+    # Setup required variables
+    root = root or get_project_root(Path(os.path.abspath('.')))
+
+    if url and (repo_name or repo_owner):
+        repo_name, repo_owner = None, None
+
+    if not url:
+        if repo_name is not None and repo_owner is not None:
+            url = urllib.parse.urljoin(host, f'{repo_owner}/{repo_name}')
+        else:
+            for remote in git.Repo(root).remotes:
+                if host in remote.url:
+                    url = remote.url[:-4]
+    elif url[-4] == '.':
+        url = url[:-4]
+
+    if not (repo_name and repo_owner):
+        splitter = lambda x: (x[-1], x[-2]) # noqa E721
+        repo_name, repo_owner = splitter(url.split('/'))
+
+    if None in [repo_name, repo_owner, url]:
+        raise ValueError('Could not parse repository owner and name. Make sure you specify either a link \
+                to the repository with --url or a pair of --repo-owner and --repo-name')
+
+    # Setup authentication
+    username = config.username
+    password = config.password
+    if username is not None and password is not None:
+        auth = username, password
+    else:
+        token = config.token or get_token()
+        if token is not None:
+            auth = token, token
+            bearer = HTTPBearerAuth(token)
+
+    # Configure repository
+    res = http_request("GET", urllib.parse.urljoin(host, config.REPO_INFO_URL.format(
+        owner=repo_owner,
+        reponame=repo_name)), auth=bearer or auth)
+    if res.status_code == 404:
+        create_repo(repo_name)
+
+    # Configure MLFlow
+    if mlflow:
+        os.environ['MLFLOW_TRACKING_URI'] = f'{url}.mlflow'
+        os.environ['MLFLOW_TRACKING_USERNAME'] = auth[0]
+        os.environ['MLFLOW_TRACKING_PASSWORD'] = auth[1]
+
+    # Configure DVC
+    if dvc:
+        Path(root / '.dvc').mkdir(parents=True, exist_ok=True)
+        write = True
+
+        dvc_config = configparser.ConfigParser()
+        dvc_config_local = configparser.ConfigParser()
+        dvc_config.read(root / '.dvc' / 'config')
+        dvc_config_local.read(root / '.dvc' / 'config.local')
+
+        if 'core' not in dvc_config.sections():
+            dvc_config['core'] = {'analytics': False,
+                                  'autostage': True}
```

Comment:
I think we shouldn't be enforcing our opinion on what dvc features to toggle on the users, autostage specifically is very much a surprising change.

Reply:
I'll clear autostage, but I think analytics being active by default is an unethical practice, so I'm in strong support for that to stay. 

Can we keep that, or should I remove it too?